pkg/test/istio.io/tasks/security/authn-policy/part1-configure-destinationrule-httpbin-legacy.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
```

Comment:
I think we need some mechanism to not have these first two lines show up in the output that's generated for use on istio.io. One option is to not include the lines in these files, and instead have the  test  framework insert them autoamtically before executing the script. The alternative is to automatically remove these lines when producing the final output.

Reply:
In this particular case, I can replace this with bash's `-e` command-line parameter which has the same effect:
```
#!/bin/bash -e
```

We could then filter out all comments (any line matching `^#`).

We also need a way to capture the output of commands, both for asserting on it in tests, and for documentation. Many examples in the docs show the output of `curl`, for instance. I think the script's expected output should be contained in the script itself, in comments, similarly to [Python doctest](https://docs.python.org/2/library/doctest.html). Maybe something like:

```
#!/bin/bash -e
# expect exit status 22
# expect output begin
# 401
# expect output end

kubectl exec $(kubectl get pod -l app=sleep -n legacy -o jsonpath={.items..metadata.name}) -c sleep -n legacy -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
```

This doesn't allow capturing the output per command, but it's close to what we need.

Reply:
Having the test automatically insert them sounds good to me.